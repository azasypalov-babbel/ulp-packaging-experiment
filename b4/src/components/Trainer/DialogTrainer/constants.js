export const TIMEOUT_AFTER_CONTEXTUAL_INFO = 2000;
