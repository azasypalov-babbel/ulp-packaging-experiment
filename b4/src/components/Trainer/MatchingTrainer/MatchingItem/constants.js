import { rem } from 'polished';

export const APPEARANCE_TYPE = {
  DEFAULT: 'DEFAULT',
  ACTIVE: 'ACTIVE',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR'
};

export const itemPairMargin = rem('16px');
export const horizontalTravel = '6rem';
