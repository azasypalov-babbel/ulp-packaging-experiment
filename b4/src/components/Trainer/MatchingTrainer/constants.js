export const MATCHING_TRAINER_STATE = {
  READY: 'READY',
  PLAYING_AUDIO: 'PLAYING_AUDIO',
  AWAITING_CONTINUE: 'AWAITING_CONTINUE',
  DONE: 'DONE',
  FEEDBACK: 'FEEDBACK'
};

export const ITEM_FRAGMENT_TYPE = {
  BASE: 'BASE',
  OPTION: 'OPTION'
};

export const trainerMaxWidth = '56rem';
