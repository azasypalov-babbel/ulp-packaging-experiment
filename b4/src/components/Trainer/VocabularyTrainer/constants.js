export const TRAINER_STATE = {
  READY: 'READY',
  AWAITING_SOUND_PLAY: 'AWAITING_SOUND_PLAY',
  AWAITING_CONTINUE: 'AWAITING_CONTINUE',
  DONE: 'DONE'
};
