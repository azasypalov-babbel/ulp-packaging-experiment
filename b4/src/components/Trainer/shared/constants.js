export const PLAYBACK_RATE_NORMAL = 1;
export const PLAYBACK_RATE_SLOW = 0.7;
