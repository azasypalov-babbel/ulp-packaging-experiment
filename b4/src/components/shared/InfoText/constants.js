export const TIMEOUT_AFTER_INFOTEXT_DISPLAY = 1000;
