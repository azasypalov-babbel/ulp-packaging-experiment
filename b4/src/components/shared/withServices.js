import React, { Component } from 'react';
import { getDisplayName } from '../getDisplayName';

export const ServiceContext = React.createContext({});

const pick = (obj, arr) =>
  arr.reduce((acc, curr) => (curr in obj && (acc[curr] = obj[curr]), acc), {});

export const withServices = (serviceNames) => (WrappedComponent) => {
  const WithServices = (props) => (
    <ServiceContext.Consumer>
      {(services) => (
        <WrappedComponent
          {...pick(services, serviceNames)}
          {...props}
        />
      )}
    </ServiceContext.Consumer>
  );

  WithServices.displayName = `WithServices(${getDisplayName(WrappedComponent)})`;

  return WithServices;
};

export const withServicesProvider = (selectServices) => (WrappedComponent) => {
  class ServicesProvider extends Component {
    constructor(props) {
      super(props);
      this.services = selectServices();
    }
    render() {
      return (
        <ServiceContext.Provider value={this.services}>
          <WrappedComponent {...this.props} />
        </ServiceContext.Provider>
      );
    }
  }

  ServicesProvider.displayName = `ServicesProvider(${getDisplayName(WrappedComponent)})`;

  return ServicesProvider;
};
