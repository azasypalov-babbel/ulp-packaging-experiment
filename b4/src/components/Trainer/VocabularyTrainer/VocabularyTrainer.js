import PropTypes from 'prop-types';
import { markupStringToHTML } from '@lessonnine/babbel-markup-helper.js';
import React, { useCallback, useContext, useEffect, useRef } from 'react';
import withTranslations from '../../shared/withTranslations';
import StyledTrainerLayout from '../../TrainerLayout';
import TranslationVisibilityToggle from '../shared/TranslationVisibilityToggle';
import Text from '../../shared/Text';
import TrainerTitle from '../shared/TrainerTitle';

import {
  StyledImage,
  StyledTextWrapper,
  StyledWrapper
} from './styles';
import { scrollIntoSafeArea } from '../../shared/scrollIntoSafeArea';
import { TrainerPlayButton } from './TrainerPlayButton';
import { useAutoRateSoundPlayer } from '../shared/useAutoRateSoundPlayer';
import { TRAINER_STATE } from './constants';
import { ServiceContext } from '../../shared/withServices';
import { checks, PlayStates } from '../../../services/soundService';

export const VocabularyTrainer = ({
  item,
  dictate,
  trainerState,
  onSoundPlayComplete,
  onToggleTranslationVisibility,
  title,
  interaction,
  interactionType,
  translations,
  translationVisibility,
  translationVisible
}) => {
  const { mediaUrlService, focusService } = useContext(ServiceContext);
  const audioUrl = item.sound?.id && mediaUrlService.soundURL(item.sound.id);
  const imageUrl = item.image?.id && mediaUrlService.imageURL(
    item.image.id, '200x200'
  );
  const ref = useRef();
  const {
    getDuration,
    play,
    playAutoRate,
    getState,
    reset,
    isSlowPlayback
  } = useAutoRateSoundPlayer(audioUrl);
  const playState = getState ? getState() : PlayStates.STOPPED;
  const duration = getDuration ? getDuration() : 0;
  const handleClick = useCallback(() => {
    focusService.returnFocus();
    playAutoRate();
  }, [focusService, playAutoRate]);
  // if 2 subsequent items have the same url
  // after finishing one item the state remains completed
  // so we need to reset the sound state whenever we get a new one
  useEffect(() => {
    reset();
  }, [item, reset]);

  const active = trainerState === TRAINER_STATE.READY;
  const waitSound = trainerState === TRAINER_STATE.AWAITING_SOUND_PLAY;
  useEffect(() => {
    if (active) {
      scrollIntoSafeArea(ref.current);
      if (dictate) {
        // autoplay sound on mount for dictate
        play({ skipOnFreeze: true });
      }
    }
    if (waitSound && !dictate) {
      // play sound after complete for vocab
      play({ skipOnFreeze: true });
    }
  }, [trainerState, dictate, play]);
  useEffect(() => {
    if (checks.hasPlayed(playState)) onSoundPlayComplete();
  }, [playState, onSoundPlayComplete]);

  const renderImage = () => {
    if (!imageUrl) {
      return null;
    }
    return (
      <StyledImage
        src={imageUrl}
        data-selector={audioUrl ? 'image-with-sound' : 'image-without-sound'}
        onClick={handleClick}
      />
    );
  };

  const renderSoundPlayer = () => {
    return (
      <TrainerPlayButton
        playSound={handleClick}
        isPlaying={playState === PlayStates.PLAYING}
        isSlowPlayback={isSlowPlayback}
        duration={duration}
      />
    );
  };

  const itemTranslation = <Text
    data-selector="item-translation"
    as="p"
    className="item-translation"
    color="spaceGrayW28"
    fontSize="base"
    dangerouslySetInnerHTML={{ __html: markupStringToHTML(item.displayLanguageText) }}
  />;

  const isCompleted = trainerState !== TRAINER_STATE.READY;
  const translationVisibleWithDictate = dictate ? isCompleted && translationVisible : translationVisible;
  const showTranslation = translationVisibility === 'partial' ? translationVisible : translationVisibleWithDictate;
  const trainerTitleKey = dictate ? `dictate${interactionType}` : interactionType;
  // vocabulary trainer has only phrase items
  return (
    <>
      <TranslationVisibilityToggle
        translationVisibility={translationVisibility}
        onClick={onToggleTranslationVisibility}
      />
      <StyledTrainerLayout ref={ref}>
        <StyledWrapper data-selector={`vocabulary-${interactionType}`}>
          <TrainerTitle text={title || translations.trainerTitle[trainerTitleKey]} />
          {dictate ? renderSoundPlayer() : renderImage()}
          <StyledTextWrapper>
            <div data-item-type="phrase">
              <div data-selector="interactive">
                {interaction}
              </div>
            </div>
          </StyledTextWrapper>
          {showTranslation && itemTranslation}
        </StyledWrapper>
      </StyledTrainerLayout>
    </>
  );
};

VocabularyTrainer.propTypes = {
  item: PropTypes.object.isRequired,
  dictate: PropTypes.bool.isRequired,
  trainerState: PropTypes.oneOf(Object.values(TRAINER_STATE)).isRequired,
  translationVisibility: PropTypes.string.isRequired,
  translations: PropTypes.shape({
    trainerTitle: PropTypes.shape({
      fillin: PropTypes.string.isRequired,
      puzzlehelper: PropTypes.string.isRequired,
      wordorder: PropTypes.string.isRequired
    })
  }).isRequired,
  title: PropTypes.string,
  interactionType: PropTypes.string.isRequired,
  interaction: PropTypes.node.isRequired,
  onSoundPlayComplete: PropTypes.func.isRequired,
  onToggleTranslationVisibility: PropTypes.func.isRequired,
  translationVisible: PropTypes.bool.isRequired
};

const getTranslations = (translate) => {
  const writeTranslationKey = 'b3.page_vocabulary_gap.type_the_missing_characters';
  return {
    trainerTitle: {
      dictatefillin: translate('b3.page_dictate.title'),
      dictatepuzzlehelper: translate('b3.page_dictate.title'),
      dictatechoose: translate('b3.page_dictate.choice_title'),
      dictatewordorder: translate('b3.page_dictate.word_order_title'),
      fillin: translate(writeTranslationKey),
      puzzlehelper: translate(writeTranslationKey),
      choose: translate('b3.page_card.title_choicebuttons'),
      wordorder: translate('b3.vocabulary_word_order.click_on_the_parts_of_the_sentence_in_the_correct_order')
    }
  };
};

export default withTranslations(getTranslations)(VocabularyTrainer);
