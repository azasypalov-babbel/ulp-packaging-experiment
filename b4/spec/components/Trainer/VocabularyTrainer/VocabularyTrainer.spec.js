import React from 'react';
import { act } from 'react-dom/test-utils';
import { mountWithTheme } from '../../shared/themeMock';

import { VocabularyTrainer } from '../../../../src/components/Trainer/VocabularyTrainer/VocabularyTrainer';
import { TRAINER_STATE } from '../../../../src/components/Trainer/VocabularyTrainer/constants';
import { withServicesProvider } from '../../../../src/components/shared/withServices';
import mockSoundService from '../../../../src/services/soundService';

jest.mock('../../../../src/services/soundService');
jest.mock('../../../../src/lib/matchingUtils/normalise');

const MockInteraction = () => null;

describe('<VocabularyTrainer />', () => {
  const item = {
    id: '123',
    sound: { id: '123' },
    image: { id: '123' },
    learnLanguageText: 'Frage "nach" einer Person mit ((wer)). ((Wer)) **bist du**?',
    displayLanguageText: 'Ask "for" a person with who. Who **are you**?',
    infoText: 'This is a "formatted" info **text**'
  };

  const defaultProps = {
    dictate: false,
    translations: {
      trainerTitle: {
        dictatefillin: 'Vocabulary DictateFillin Title',
        dictatepuzzlehelper: 'Vocabulary DictatePuzzlehelper Title',
        dictatewordorder: 'Vocabulary DictateWordorder Title',
        fillin: 'Vocabulary Fillin Title',
        puzzlehelper: 'Vocabulary Puzzlehelper Title',
        wordorder: 'Vocabulary Wordorder Title'
      }
    },
    title: null,
    item,
    interaction: <MockInteraction />,
    interactionType: 'fillin',
    onToggleTranslationVisibility: jest.fn(),
    onSoundPlayComplete: () => {},
    trainerState: TRAINER_STATE.READY,
    currentIndex: 0,
    translationVisible: true,
    translationVisibility: 'full'
  };
  const services = {
    mediaUrlService: {
      soundURL: jest.fn((url) => `mock-url-${url}.mp3`),
      imageURL: jest.fn((url) => `mock-url-${url}.jpg`)
    },
    soundService: mockSoundService
  };
  const Component = withServicesProvider(() => services)(VocabularyTrainer);

  const getWrapper = (props = {}) => mountWithTheme(
    <Component {...defaultProps} {...props} />
  );
  describe('renders', () => {
    describe('images', () => {
      it('with soundless items', async () => {
        const soudlessItem = { sound: null, ...item };
        const wrapper = getWrapper({ item: soudlessItem });
        expect(wrapper.find('img[src="mock-url-123.jpg"]').exists()).toBe(true);
        expect(wrapper.find('PlayButton').exists()).toBe(false);
      });
    });

    describe('sound player', () => {
      it('with default props', async () => {
        const wrapper = getWrapper({ dictate: true });
        expect(wrapper.find('PlayButton').exists()).toBe(true);
        expect(wrapper.find('img[src="mock-url-123.jpg"]').exists()).toBe(false);
      });

      it('with soundless items', async () => {
        const soudlessItem = { sound: null, ...item };
        const wrapper = getWrapper({ dictate: true, item: soudlessItem });
        expect(wrapper.find('PlayButton').exists()).toBe(true);
      });
    });

    it('with full props', async ()=> {
      const wrapper = getWrapper();
      expect(wrapper).toMatchSnapshot();
    });

    it('with an interaction', async () => {
      const wrapper = getWrapper();
      expect(wrapper.exists(MockInteraction)).toBe(true);
    });
  });

  describe('interaction', () => {
    describe('automatically playing the sound', () => {
      it('dictate plays sound on mount', () => {
        getWrapper({ dictate: true });
        const soundInstanceMock = services.soundService.getInstance('mock-url-123.mp3');
        expect(soundInstanceMock.play).toBeCalledTimes(1);
      });

      it('vocab plays sound on mount', () => {
        const wrapper = getWrapper({ dictate: false });
        const soundInstanceMock = services.soundService.getInstance('mock-url-123.mp3');
        expect(soundInstanceMock.play).toBeCalledTimes(0);
        wrapper.setProps({ trainerState: TRAINER_STATE.AWAITING_CONTINUE });
        wrapper.update();
      });
    });

    describe('manually playing sound', () => {
      it('play the sound when play button pressed', async () =>{
        const soundInstanceMock = services.soundService.getInstance('mock-url-123.mp3');
        const wrapper = getWrapper({ dictate: true });
          // end autoplay first
        await act(async () => {
          soundInstanceMock.simulate('end');
        });
        wrapper.update();
        expect(wrapper.find('PlayButton').prop('isPlaying')).toBe(false);
        await act(async () => {
          soundInstanceMock.simulate('play');
          wrapper.find('PlayButton').invoke('onClick');
        });
        wrapper.update();
        expect(wrapper.find('PlayButton').prop('isPlaying')).toBe(true);
      });
    });
  });

  describe('title', () => {
    describe('when prop title is defined', () => {
      it('should render title from props', async () => {
        const titleText = 'test title';
        const wrapper = getWrapper({ title: titleText });
        expect(wrapper.find('[data-selector="title"]').find('h1').text())
          .toEqual(titleText);
      });
    });

    describe('when prop title is not defined', () => {
      it('should render title from translations based on the trainer interaction', async () => {
        const wrapper = getWrapper();
        expect(wrapper.find('[data-selector="title"]').find('h1').text())
          .toEqual(defaultProps.translations.trainerTitle[defaultProps.interactionType]);
      });

      it('should render title from dictate translations based on the trainer interaction', async () => {
        const wrapper = getWrapper({ dictate: true });
        expect(wrapper.find('[data-selector="title"]').find('h1').text())
          .toEqual(defaultProps.translations.trainerTitle[`dictate${defaultProps.interactionType}`]);
      });
    });
  });

  describe('title formatting', () => {
    it('is called, if there is a title', async () => {
      const title = 'formatt{ed}';
      const wrapper = getWrapper({ title });
      expect(wrapper.find('[data-selector="title"]').find('h1').text()).toEqual('formatt(ed)');
    });

    it('can handle html / babbel markup formatting', async () => {
      const title = 'a "really" nicely **formatted** title';
      const textTitle = 'a really nicely formatted title';
      const formattedTitle = 'a <i>really</i> nicely <b>formatted</b> title';
      const wrapper = getWrapper({ title });
      expect(wrapper.find('[data-selector="title"]').find('h1').text()).toEqual(textTitle);
      expect(wrapper.find('[data-selector="title"]').find('h1').html()).toContain(formattedTitle);
    });
  });

  describe('Translations', () => {
    describe('with "full" translation visibility', () => {
      it('should display translation text', async () => {
        expect(
          getWrapper().find('[data-selector="item-translation"]').first().text()
        ).toEqual('Ask for a person with who. Who are you?');
      });

      it('should not display a toggle button', async () => {
        expect(getWrapper().exists('[data-selector="translation-toggle"]')).toBe(false);
      });

      it('should support html format for translation text', async () => {
        expect(
          getWrapper().find('[data-selector="item-translation"]').first().html()
        ).toContain('Ask <i>for</i> a person with who. Who <b>are you</b>?');
      });
    });

    describe('with "none" translation visibility', () => {
      const visibilityProps = {
        translationVisibility: 'none',
        translationVisible: false
      };
      it('should not display translation text', () => {
        expect(getWrapper(visibilityProps).exists('[data-selector="item-translation"]')).toBe(false);
      });

      it('should not display a toggle button', () => {
        expect(getWrapper(visibilityProps).exists('[data-selector="translation-toggle"]')).toBe(false);
      });
    });

    describe('with "partial" translation visibility', () => {
      const visibilityProps = {
        translationVisibility: 'partial',
        translationVisible: false
      };
      it('should not display translation text', async () => {
        expect(getWrapper(visibilityProps).exists('[data-selector="item-translation"]')).toBe(false);
      });

      it('should display a toggle button', async () => {
        expect(getWrapper(visibilityProps).exists('[data-selector="translation-toggle"]')).toBe(true);
      });

      it('should execute toggle function when clicking on the toggle button', async () => {
        getWrapper(visibilityProps).find('[data-selector="translation-toggle"]').at(4).simulate('click');
        expect(defaultProps.onToggleTranslationVisibility).toHaveBeenCalled();
      });
    });
  });

  describe('review item with no image or sound', () => {
    const item = {
      id: '3920bdb267b8a8044e013c845ba23973',
      displayLanguageText: 'Das ist die Gelegenheit, uns kennenzulernen!',
      learnLanguageText: '((*C\'est l\'occasion|*C\'est le moment|*C\'est l\'opportunitÃ©)) de nous rencontrer !',
      info_text: '',
      type: 'phrase',
      image: null,
      sound: null
    };

    it('shound render without an image', async () => {
      const wrapper = getWrapper({ item });
      expect(wrapper.exists('img')).toBe(false);
    });
  });
});
