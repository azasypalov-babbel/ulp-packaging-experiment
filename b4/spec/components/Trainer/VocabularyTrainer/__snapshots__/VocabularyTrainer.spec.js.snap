// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VocabularyTrainer /> renders with full props 1`] = `
.c0 {
  margin: 1rem auto 1rem;
  margin: clamp(1rem,6vw,4rem) auto 1rem;
}

.c2 {
  font-family: Milliard-SemiBold,sans-serif;
  color: #394149;
  font-weight: normal;
  line-height: 1.25;
  -webkit-letter-spacing: 0.031rem;
  -moz-letter-spacing: 0.031rem;
  -ms-letter-spacing: 0.031rem;
  letter-spacing: 0.031rem;
  text-align: center;
}

.c2 > strong,
.c2 > b {
  font-family: Milliard-SemiBold,sans-serif;
  font-weight: normal;
}

.c2 > em,
.c2 > i {
  font-family: Milliard-BookItalic,sans-serif;
  font-weight: normal;
  font-style: normal;
}

.c2 > strong em,
.c2 > b i {
  font-family: Milliard-MediumItalic,sans-serif;
  font-weight: normal;
}

.c7 {
  font-family: Milliard-Book,sans-serif;
  color: #70767c;
  font-weight: normal;
  line-height: 1.25;
  -webkit-letter-spacing: 0.031rem;
  -moz-letter-spacing: 0.031rem;
  -ms-letter-spacing: 0.031rem;
  letter-spacing: 0.031rem;
  text-align: inherit;
}

.c7 > strong,
.c7 > b {
  font-family: Milliard-SemiBold,sans-serif;
  font-weight: normal;
}

.c7 > em,
.c7 > i {
  font-family: Milliard-BookItalic,sans-serif;
  font-weight: normal;
  font-style: normal;
}

.c7 > strong em,
.c7 > b i {
  font-family: Milliard-MediumItalic,sans-serif;
  font-weight: normal;
}

.c3 {
  margin-bottom: 2rem;
}

.c3 > b,
.c3 > i {
  font-family: Milliard-MediumItalic,sans-serif;
}

.c4 {
  margin-left: 2.25rem;
  margin-right: 2.25rem;
  margin-bottom: rem(24);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
  margin: 0 auto;
  max-width: rem(640);
  width: 100%;
  box-sizing: border-box;
}

.c6 {
  font-size: 1.5rem;
  margin-bottom: 1rem;
}

.c5 {
  margin-bottom: 1.5rem;
  border-radius: 0.5rem;
  width: 140px;
  height: 140px;
}

@media (min-width:360px) {
  .c4 {
    margin-bottom: rem(32);
  }
}

@media (min-width:600px) {
  .c4 {
    margin-bottom: rem(48);
  }
}

@media (min-width:768px) {
  .c5 {
    width: 200px;
    height: 200px;
  }
}

<ServicesProvider(VocabularyTrainer)
  currentIndex={0}
  dictate={false}
  interaction={<MockInteraction />}
  interactionType="fillin"
  item={
    Object {
      "displayLanguageText": "Ask \\"for\\" a person with who. Who **are you**?",
      "id": "123",
      "image": Object {
        "id": "123",
      },
      "infoText": "This is a \\"formatted\\" info **text**",
      "learnLanguageText": "Frage \\"nach\\" einer Person mit ((wer)). ((Wer)) **bist du**?",
      "sound": Object {
        "id": "123",
      },
    }
  }
  onSoundPlayComplete={[Function]}
  onToggleTranslationVisibility={[MockFunction]}
  title={null}
  trainerState="READY"
  translationVisibility="full"
  translationVisible={true}
  translations={
    Object {
      "trainerTitle": Object {
        "dictatefillin": "Vocabulary DictateFillin Title",
        "dictatepuzzlehelper": "Vocabulary DictatePuzzlehelper Title",
        "dictatewordorder": "Vocabulary DictateWordorder Title",
        "fillin": "Vocabulary Fillin Title",
        "puzzlehelper": "Vocabulary Puzzlehelper Title",
        "wordorder": "Vocabulary Wordorder Title",
      },
    }
  }
>
  <VocabularyTrainer
    currentIndex={0}
    dictate={false}
    interaction={<MockInteraction />}
    interactionType="fillin"
    item={
      Object {
        "displayLanguageText": "Ask \\"for\\" a person with who. Who **are you**?",
        "id": "123",
        "image": Object {
          "id": "123",
        },
        "infoText": "This is a \\"formatted\\" info **text**",
        "learnLanguageText": "Frage \\"nach\\" einer Person mit ((wer)). ((Wer)) **bist du**?",
        "sound": Object {
          "id": "123",
        },
      }
    }
    onSoundPlayComplete={[Function]}
    onToggleTranslationVisibility={[MockFunction]}
    title={null}
    trainerState="READY"
    translationVisibility="full"
    translationVisible={true}
    translations={
      Object {
        "trainerTitle": Object {
          "dictatefillin": "Vocabulary DictateFillin Title",
          "dictatepuzzlehelper": "Vocabulary DictatePuzzlehelper Title",
          "dictatewordorder": "Vocabulary DictateWordorder Title",
          "fillin": "Vocabulary Fillin Title",
          "puzzlehelper": "Vocabulary Puzzlehelper Title",
          "wordorder": "Vocabulary Wordorder Title",
        },
      }
    }
  >
    <TranslationVisibilityToggle
      onClick={[MockFunction]}
      translationVisibility="full"
    />
    <styled.div>
      <div
        className="c0"
      >
        <styled.div
          data-selector="vocabulary-fillin"
        >
          <div
            className="c1"
            data-selector="vocabulary-fillin"
          >
            <TrainerTitle
              text="Vocabulary Fillin Title"
            >
              <Styled(Title)
                as="h1"
                color="spaceGray"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Vocabulary Fillin Title",
                  }
                }
                data-selector="title"
                fontFamily="fontMilliardSemi"
                fontSize="big"
                textAlign="center"
              >
                <h1
                  className="c2 c3 c4"
                  color="spaceGray"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Vocabulary Fillin Title",
                    }
                  }
                  data-selector="title"
                  fontFamily="fontMilliardSemi"
                  fontSize="big"
                />
              </Styled(Title)>
            </TrainerTitle>
            <styled.img
              data-selector="image-with-sound"
              onClick={[Function]}
              src="mock-url-123.jpg"
            >
              <img
                className="c5"
                data-selector="image-with-sound"
                onClick={[Function]}
                src="mock-url-123.jpg"
              />
            </styled.img>
            <styled.div>
              <div
                className="c6"
              >
                <div
                  data-item-type="phrase"
                >
                  <div
                    data-selector="interactive"
                  >
                    <MockInteraction />
                  </div>
                </div>
              </div>
            </styled.div>
            <styled.span
              as="p"
              className="item-translation"
              color="spaceGrayW28"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Ask <i>for</i> a person with who. Who <b>are you</b>?",
                }
              }
              data-selector="item-translation"
              fontSize="base"
            >
              <p
                className="c7 item-translation"
                color="spaceGrayW28"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Ask <i>for</i> a person with who. Who <b>are you</b>?",
                  }
                }
                data-selector="item-translation"
                fontSize="base"
              />
            </styled.span>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </VocabularyTrainer>
</ServicesProvider(VocabularyTrainer)>
`;
